diff --git a/bin/nexus b/bin/nexus
--- a/bin/nexus
+++ b/bin/nexus
@@ -10,9 +10,15 @@
 # Short-Description: nexus
 ### END INIT INFO
 
+if [ -n "$ALTERNATIVE_NAME" ]; then
+  execname="$ALTERNATIVE_NAME"
+else
+  execname="$actualfile"
+fi
+
 original_dir=`pwd`
 
-actualfile=$(realpath "$0")
+actualfile=$(realpath $execname)
 progname=`basename "$actualfile"`
 linkdir=`dirname "$actualfile"`
 HOME="$linkdir"
@@ -139,7 +145,8 @@ read_vmoptions() {
 
 cd "$HOME" || exit 1
 vmoptions_val=""
-read_vmoptions "nexus.vmoptions"
+VM_OPTS=${VM_OPTS_FILE:-"nexus.vmoptions"}
+read_vmoptions "$VM_OPTS"
 INSTALL4J_ADD_VM_PARAMS="$INSTALL4J_ADD_VM_PARAMS $vmoptions_val"
 
 # deduce the chosen data directory and prepare log and tmp directories
@@ -177,7 +184,7 @@ case "$1" in
   ;;
   start-launchd)
     echo "Starting nexus"
-    $0 run
+    $actualfile run
   ;;
   stop)
     echo "Shutting down nexus"
@@ -203,8 +210,8 @@ case "$1" in
     fi
   ;;
   restart|force-reload)
-    $0 stop
-    $0 start
+    $actualfile stop
+    $actualfile start
   ;;
   status)
     # no-op
@@ -217,7 +224,7 @@ case "$1" in
     cd "$original_dir" || exit 1
   ;;
   *)
-    echo "Usage: $0 {start|stop|run|run-redirect|status|restart|force-reload}"
+    echo "Usage: $actualfile {start|stop|run|run-redirect|status|restart|force-reload}"
     exit 1
   ;;
 esac
